cms/models/managers.py

```diff
@@ -124,8 +124,16 @@ def get_title(self, page, language, language_fallback=False):
 
 
 class PageContentAdminQuerySet(models.QuerySet):
-    def current_content_iterator(self, **kwargs):
-        return iter(self.filter(**kwargs))
+    def current_content(self, **kwargs):
```

Comment:
Because these both filter on arbitrary kwargs, it doesn't seem right to have two methods and to name them like this - you could have kwargs that lead to it filtering on something other that current/latest content.

Reply:
The `kwargs` are not needed but provide a little syntactic sugar to allow for something like `PageContent.admin_manager.current_content(language="en")` wich would be identical to the longer `PageContent.admin_manager.filter(language="en").current_content()`. (Both give a qs of all "current" page contents in English - without versioning that's just all.)

Typically (but as you indicate correctly not necessarily) those kwargs would be one of the "extra grouping fields" known to versioning but unknown to the core. For page content we known it to be `language`. An alternative would be to test if `kwargs` contains anything else than a language argument and throw an exception. 

If we were to fully remove the `kwargs`, the API of djangocms-versioning would need an according change (since it adds non-trivial functionality to the method).

The kwargs-filtering is not used in the core code base but in a test which would need to be adjusted.

@marksweb , what's your take?

In any case I'll explain the existence of the methods in a docstring.

Reply:
@fsbraun I'm not talking about removing the passing kwargs. That makes absolute sense.

It's the fact that two methods both return `self.filter(**kwargs)` that seems odd. Their names suggest that one should get the latest content and the other should get the current content. But without kwargs, they don't actually do either of those things.